Add unreleased docker release notes (#33855)
sourcegraph/sourcegraph: doc/admin/updates/docker_compose.md

<!-- GENERATE UPGRADE GUIDE ON RELEASE (release tooling uses this to add entries) -->

## Unreleased

> We made a number of changes to our build-in postgres databases (the `pgsql`, `codeintel-db`, and `codeinsights-db` container)

Author: ```suggestion
We made a number of changes to our built-in postgres databases (the `pgsql`, `codeintel-db`, and `codeinsights-db` container)
```

> - **CAUTION** Add the ability to customize postgres server configuration. If you have customized the config in any way, you should copy your changes to the added `postgresql.conf` file [sourcegraph/deploy-sourcegraph-docker#792](https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/792)

Author: ```suggestion
- **CAUTION** Add the ability to customize postgres server configuration. If you have customized the config in any way, you should copy your changes to the added `postgresql.conf` files [sourcegraph/deploy-sourcegraph-docker#792](https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/792)
```

> - **CAUTION** Add the ability to customize postgres server configuration. If you have customized the config in any way, you should copy your changes to the added `postgresql.conf` files [sourcegraph/deploy-sourcegraph-docker#792](https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/792)

Reviewer: ```suggestion
- **CAUTION** Added the ability to customize postgres server configuration by mounting external configuration files. If you have customized the config in any way, you should copy your changes to the added `postgresql.conf` files [sourcegraph/deploy-sourcegraph-docker#792](https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/792)
```

> - Increase the minimal memory requirement of `pgsql` and `codeintel-db` from `2GB` to `4GB`

Reviewer: ```suggestion
- Increased the minimal memory requirement of `pgsql` and `codeintel-db` from `2GB` to `4GB`
```

> - `codeinsights-db` now uses vanilla postgres image [sourcegraph/deploy-sourcegraph-docker#780](https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/780)

Author: ```suggestion
- `codeinsights-db` container now uses vanilla postgres image [sourcegraph/deploy-sourcegraph-docker#780](https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/780)
```

> - `codeinsights-db` container now uses vanilla postgres image [sourcegraph/deploy-sourcegraph-docker#780](https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/780)

Reviewer: ```suggestion
-`codeinsights-db` container no longer uses TimescaleDB and is now based on the standard Postgres image [sourcegraph/deploy-sourcegraph-docker#780](https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/780). Metrics scraping is also enabled.
```

> -`codeinsights-db` container no longer uses TimescaleDB and is now based on the standard Postgres image [sourcegraph/deploy-sourcegraph-docker#780](https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/780). Metrics scraping is also enabled.

Author: ```suggestion
-`codeinsights-db` container no longer uses TimescaleDB and is now based on the standard Postgres image [sourcegraph/deploy-sourcegraph-docker#780](https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/780). Metrics scraping is also enabled. You may now use external Postgres deployment `codeinsights-db`
```

@caugustus-sourcegraph should we mention user can host codeinsights-db elsewhere?

Reviewer: That should be covered by the general release notes (I know Code Insights was working on a release advisory). These update docs are specifically for "things that might break your docker-compose setup".
